buildSrc/src/main/java/com/google/firebase/gradle/ReleaseGenerator.kt

```diff
@@ -13,39 +13,111 @@
 // limitations under the License.
 package com.google.firebase.gradle
 
+import com.google.common.collect.ImmutableList
+import com.google.common.collect.Sets
+import com.google.firebase.gradle.plugins.FirebaseLibraryExtension
 import java.io.File
 import org.eclipse.jgit.api.Git
 import org.eclipse.jgit.api.ListBranchCommand
 import org.eclipse.jgit.api.errors.GitAPIException
 import org.eclipse.jgit.lib.Constants
 import org.eclipse.jgit.lib.ObjectId
+import org.eclipse.jgit.revwalk.RevCommit
 import org.gradle.api.DefaultTask
+import org.gradle.api.Project
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputFile
 import org.gradle.api.tasks.TaskAction
 
 data class FirebaseLibrary(val moduleNames: List<String>, val directories: List<String>)
 
-open class ReleaseGenerator : DefaultTask() {
+data class CommitDiff(
+  val commitId: String,
+  val author: String,
+  val message: String,
+) {
+  override fun toString(): String =
+    """
+      |* ${message.split("\n").first()}   
+      |  https://github.com/firebase/firebase-android-sdk/commit/${commitId}  [${author}]
+
+    """
+      .trimMargin()
+}
+
+abstract class ReleaseGenerator : DefaultTask() {
+
+  @get:Input public abstract val currentRelease: Property<String>
+
+  @get:Input public abstract val pastRelease: Property<String>
+
+  @get:Input public abstract val printReleaseConfig: Property<String>
+
+  @OutputFile public fun getReleaseConfigFile() = project.buildDir.resolve("release.cfg")
+
+  @OutputFile public fun getReleaseReportFile() = project.buildDir.resolve("release_report.md")
+
   @TaskAction
   @Throws(Exception::class)
   fun generateReleaseConfig() {
-    val currentRelease = project.property("currentRelease").toString()
-    val pastRelease = project.property("pastRelease").toString()
-    val printReleaseConfig = project.property("printOutput").toString().toBoolean()
     val rootDir = project.rootDir
     val availableModules = parseSubProjects(rootDir)
```

Comment:
This relies on the existence of `subprojects.cfg`, and is not independent of the the project itself (it works from the root); which we generally wanna avoid- although sometimes it's not very feasible.

In this case, what if we instead checked for `firebaseLibrary` to validate the project (in place of `subproject.cfg`). Could also have a property in `firebaseLibrary` like `unreleased` which defaults to false, and unreleased sdks like auth exchange could mark it as true- to retain `subprojects.cfg` opt out functionality:

```kotlin
// something along these lines
val availableModules = project.subprojects.filter { it.plugins.hasPlugin("firebase-library") }
```

Alternatively- we let `FirebaseLibrary` register an individual task that outputs a release config at the project level- then a separate task in `MultiProjectReleasePlugin` aggregates the results (this would be the most "correct" way to do it, but would require a moderate amount of restructuring)